### DISCUSSION TITLE
Support passing `equalityFn` in the parameter of `Create`

### DISCUSSION THREAD

**Discussion by chenzn1** [pmndrs/zustand#1664] [Ideas]
```ts
import { create, shallow } from 'zustand'

const useStore = (
  set => ({ 
    count: 1, 
    incCount: state => set({ count: state.count + 1 }),
   
    total: 1, 
    incTotal: state => set({ total: state.incTotal + 1 }),
  }),
  shallow
)
```

**Comment by dai-shi** [comment]
Yeah, some people like it too: #685 

It's out of the scope of the core library.
You can either mutate store after creation (easy) or create middleware (hard).
Actually, I'm waiting for someone to develop such middleware. https://github.com/pmndrs/zustand/issues/685#issuecomment-1116080831

**Comment by chenzn1** [comment]
Thanks for answering, but i  forgot to say that `shallow` is to compare the result of the `selector` to prevent re-render, not all states. so it's a bit different with [#658 comment](https://github.com/pmndrs/zustand/issues/685#issuecomment-1116080831).
The middware seems unable to get the result of the `selector`?

**Reply by dai-shi** [reply]
right, middleware can only care about vanilla store.

**Comment by chenzn1** [comment]
Here's a nice solution. [#658 comment](https://github.com/pmndrs/zustand/issues/685#issuecomment-1212359370)

**Reply by dai-shi** [reply]
great if that works.
